test/integration/component/test_ospf.py

```diff
@@ -0,0 +1,477 @@
+      
+        
+        self.superman_aclitem2 =  NetworkACL.create( self.apiclient, aclid=self.superman_aclgroup.id, services=self.services["natrule"],
+                                        traffictype='Ingress' )
+        self.assertIsNotNone(self.superman_aclitem2, "Network failed to aclItem")
+        self.logger.debug("Added a network ACL %s to ACL list %s" % (self.superman_aclitem2.id, self.superman_aclgroup.name))
+        
+        """
+         57 vpc2_tier1_name=robin-tier-1
+         58 vpc2_tier1_display_name='"Robin Tier 1"'
+         59 vpc2_tier1_gateway=200.100.5.1
+         60 vpc2_tier1_netmask=255.255.255.0
+         61 
+         """
+           
+        #create tiers
+        self.superman_network_tier1 = Network.create(
+            self.apiclient,
+            self.services["superman_network_tier1"],
+            networkofferingid=self.network_offering.id,
+            zoneid=self.zone.id,
+            aclid=self.superman_aclgroup.id,
+            gateway='200.100.5.1',
+            vpcid=self.superman_vpc.id
+        )
+        self.validate_network(self.superman_network_tier1, self.superman_vpc.id)
+        self.logger.debug("Created superman_network_tier1 with ID: %s" % self.superman_network_tier1.id)
+        
+        """---------------------------------------------------------"""
+        
+        #create VMs
+        self.template = get_template(
+            self.apiclient,
+            self.zone.id,
+            template_name = self.services["template_name"]
+        )
+        
+        self.logger.debug("Using template %s " % self.template.id)
+                
+        self.service_offering = ServiceOffering.create(
+            self.apiclient,
+            self.services["service_offering"]
+        )
+        self.logger.debug("Using service offering %s " % self.service_offering.id)
+
+        self.services["batman_vm_1"]["zoneid"] = self.zone.id
+        self.services["batman_vm_1"]["template"] = self.template.id
+        self.batman_vm_1 = VirtualMachine.create(
+            self.apiclient,
+            self.services["batman_vm_1"],
+            serviceofferingid=self.service_offering.id,
+            ipaddress="200.100.1.101",
+            networkids=self.batman_network_tier1.id
+        )
+        self.logger.debug("Deployed VM %s in network: %s" % (self.batman_vm_1.id, self.batman_network_tier1.id))
+        
+        self.services["superman_vm_1"]["zoneid"] = self.zone.id
+        self.services["superman_vm_1"]["template"] = self.template.id
+        self.superman_vm_1 = VirtualMachine.create(
+            self.apiclient,
+            self.services["superman_vm_1"],
+            serviceofferingid=self.service_offering.id,
+            ipaddress="200.100.5.101",
+            networkids=self.superman_network_tier1.id
+        )
+        self.logger.debug("Deployed VM %s in network: %s" % (self.superman_vm_1.id, self.superman_network_tier1.id))
```

Comment:
Lines 404-413 and lines 415-424 appear similar.  Consider extracting to a private function to reduce the complexity of this test method.